lib/AST/Type.cpp

```diff
@@ -5068,6 +5068,34 @@ makeFunctionType(ArrayRef<AnyFunctionType::Param> params, Type retTy,
   return FunctionType::get(params, retTy);
 }
 
+// Determine whether the static self type of the transpose method is the
+// same as the one in the result. This should only be called when we are
+// transposing W.R.T. 'self'.
+bool AnyFunctionType::transposeSelfTypesMatch(Type *staticSelfType,
```

Comment:
Since `AnyFunctionType::transposeSelfTypesMatch` is used only in `TypeCheckAttr.cpp`, can you please change it to be a `static` (fileprivate) function, rather than defining it to `include/swift/AST/Types.h`?